src/lib/dt_extensions/dt_extensions.py

```diff
@@ -22,7 +25,10 @@ async def prepare_services_config_for_next_polling(current_services: List[GCPSer
             extensions_fetch_result = await extensions_fetch(gcp_session, dt_session, token)
             if not extensions_fetch_result:
                 raise Exception('Extension fetch failed')
-
+            
+            if config.metric_autodiscovery():
```

Comment:
return await enrich_services_autodiscovery(extensions_fetch_result,gcp_session,dt_session,token) if config.metric_autodiscovery()

Reply:
I don't think we should change it like this, autodiscovery is kind of a additional feature and not really relevant for main return. Just because we can squash everything to one line does not mean we should!